File: katt/vanner/submissions/accepted/vanner.py
pos = set()

sc = 0

def init(N, L, P):
    for i in P:
        jump(-100, i)

def jump(A, B):
    global sc
    if A-1 in pos or A+1 in pos: sc -= 1
    if A-1 in pos and A-2 not in pos: sc -= 1
    if A+1 in pos and A+2 not in pos: sc -= 1
    if A in pos: pos.remove(A)

    if B-1 in pos or B+1 in pos: sc += 1
    if B-1 in pos and B-2 not in pos: sc += 1
    if B+1 in pos and B+2 not in pos: sc += 1
    pos.add(B)

def score():
    return sc

File: katt/vanner/submissions/accepted/vanner.cpp
#include "vanner.h"
#include <set>

using namespace std;

int sc;
set<int> pos;

void init(int N, int L, int P[]) {
  for (int i = 0; i < N; ++i) jump(-100, P[i]);
}

bool in(int A) {
  return pos.find(A) != pos.end();
}

void jump(int A, int B) {
  if (in(A - 1) or in(A + 1)) sc--;
  if (in(A - 1) and not in(A - 2)) sc--;
  if (in(A + 1) and not in(A + 2)) sc--;
  pos.erase(A);

  if (in(B - 1) or in(B + 1)) sc++;
  if (in(B - 1) and not in(B - 2)) sc++;
  if (in(B + 1) and not in(B + 2)) sc++;
  pos.insert(B);
}

int score(){
  return sc;
}
